Build on successful pipeline events

## Issue

### Context
- **Gitlab plugin version**: 1.5.3
- **Gitlab version**: 9.1.4
- **Jenkins version**: 2.89.4
- **Job type**: Pipeline

### Logs & Traces

Nothing appears in the logs when I perform this action.

### Problem description

_Describe your problem in a meaningful way_:
When I set  *Build on successful pipeline events* on my build trigger and save the job it is unset when I re-open the job.
- what was expected
The setting would persist and my build would be triggered by pipeline events.
- what occurred finally
The setting does not persist.

If I modify the XML by hand then I can get the PipelineHook to trigger but unfortunately the event seems to be missing some required information. (See logs below)

I set the loglevel for `com.dabsquared.gitlabjenkins` to `fine` but this did not produce any additional log output on subsqequent runs.
```
Caused by: java.lang.NullPointerException: sourceProjectId must not be null.
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:204)
        at com.dabsquared.gitlabjenkins.cause.CauseData.<init>(CauseData.java:71)
        at com.dabsquared.gitlabjenkins.cause.CauseDataBuilder.build(CauseDataBuilder.java:632)
        at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.retrieveCauseData(PipelineHookTriggerHandlerImpl.java:122)
        at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.retrieveCauseData(PipelineHookTriggerHandlerImpl.java:32)
        at com.dabsquared.gitlabjenkins.trigger.handler.AbstractWebHookTriggerHandler.createActions(AbstractWebHookTriggerHandler.java:85)
        at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.handle(PipelineHookTriggerHandlerImpl.java:69)
        at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.handle(PipelineHookTriggerHandlerImpl.java:32)
        at com.dabsquared.gitlabjenkins.GitLabPushTrigger.onPost(GitLabPushTrigger.java:426)
        at com.dabsquared.gitlabjenkins.webhook.build.PipelineBuildAction$1.performOnPost(PipelineBuildAction.java:62)
        at com.dabsquared.gitlabjenkins.webhook.build.BuildWebHookAction$TriggerNotifier.run(BuildWebHookAction.java:54)
        at hudson.security.ACL.impersonate(ACL.java:243)
        at com.dabsquared.gitlabjenkins.webhook.build.PipelineBuildAction.execute(PipelineBuildAction.java:59)
        at com.dabsquared.gitlabjenkins.webhook.build.BuildWebHookAction.execute(BuildWebHookAction.java:31)
        at com.dabsquared.gitlabjenkins.webhook.GitLabWebHook.getDynamic(GitLabWebHook.java:44)
        at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
        at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:343)
        ... 74 more
```

@omehegan

Context
Gitlab plugin version: 1.5.10
Gitlab version: 11.2
Jenkins version: 2.138.1
Job type: Pipeline

also same problem when enable "Build on successful pipeline events"
```
Caused by: java.lang.NullPointerException: sourceProjectId must not be null.
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:204)
	at com.dabsquared.gitlabjenkins.cause.CauseData.<init>(CauseData.java:74)
	at com.dabsquared.gitlabjenkins.cause.CauseDataBuilder.build(CauseDataBuilder.java:632)
	at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.retrieveCauseData(PipelineHookTriggerHandlerImpl.java:128)
```

Same issue like this, has anyone to look it?
@omehegan @mreichel 

Gitlab plugin version: 1.5.11
Gitlab version: 11.9.1 (86f0b5dd)
Jenkins version: 2.164.1

With `Build on successful pipeline events` option checked in jenkins, after the pipeline of gitlab successful completed, the logs of jenkins's system like below:
```
Apr 03, 2019 8:20:55 AM com.dabsquared.gitlabjenkins.webhook.GitLabWebHook getDynamic
INFO: WebHook called with url: /project/Test
Apr 03, 2019 8:20:55 AM com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl handle
INFO: Test triggered for pipeline event.
Apr 03, 2019 8:20:55 AM org.eclipse.jetty.server.handler.ContextHandler$Context log
WARNING: Error while serving http://jenkins.tt/project/Test
java.lang.reflect.InvocationTargetException
	at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:400)
	at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:408)
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:212)
	at org.kohsuke.stapler.MetaClass$9.dispatch(MetaClass.java:456)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:739)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:870)
	at org.kohsuke.stapler.MetaClass$9.dispatch(MetaClass.java:458)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:739)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:870)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:668)
	at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:154)
	at jenkins.telemetry.impl.UserLanguages$AcceptLanguageFilter.doFilter(UserLanguages.java:128)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:157)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at com.dabsquared.gitlabjenkins.webhook.GitLabWebHook$GitlabWebHookCrumbExclusion.process(GitLabWebHook.java:53)
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:73)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
	at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:117)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:135)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:93)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
	at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:90)
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:503)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: sourceProjectId must not be null.
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:204)
	at com.dabsquared.gitlabjenkins.cause.CauseData.<init>(CauseData.java:74)
	at com.dabsquared.gitlabjenkins.cause.CauseDataBuilder.build(CauseDataBuilder.java:632)
	at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.retrieveCauseData(PipelineHookTriggerHandlerImpl.java:128)
	at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.retrieveCauseData(PipelineHookTriggerHandlerImpl.java:32)
	at com.dabsquared.gitlabjenkins.trigger.handler.AbstractWebHookTriggerHandler.createActions(AbstractWebHookTriggerHandler.java:85)
	at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.handle(PipelineHookTriggerHandlerImpl.java:70)
	at com.dabsquared.gitlabjenkins.trigger.handler.pipeline.PipelineHookTriggerHandlerImpl.handle(PipelineHookTriggerHandlerImpl.java:32)
	at com.dabsquared.gitlabjenkins.GitLabPushTrigger.onPost(GitLabPushTrigger.java:481)
	at com.dabsquared.gitlabjenkins.webhook.build.PipelineBuildAction$1.performOnPost(PipelineBuildAction.java:76)
	at com.dabsquared.gitlabjenkins.webhook.build.BuildWebHookAction$TriggerNotifier.run(BuildWebHookAction.java:54)
	at hudson.security.ACL.impersonate(ACL.java:273)
	at com.dabsquared.gitlabjenkins.webhook.build.PipelineBuildAction.execute(PipelineBuildAction.java:73)
	at com.dabsquared.gitlabjenkins.webhook.build.BuildWebHookAction.execute(BuildWebHookAction.java:31)
	at com.dabsquared.gitlabjenkins.webhook.GitLabWebHook.getDynamic(GitLabWebHook.java:44)
	at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
	at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:396)
	... 75 more
```
The keyword `sourceProjectId must not be null` .

--- a/src/main/java/com/dabsquared/gitlabjenkins/trigger/handler/pipeline/PipelineHookTriggerHandlerImpl.java
+++ b/src/main/java/com/dabsquared/gitlabjenkins/trigger/handler/pipeline/PipelineHookTriggerHandlerImpl.java
@@ -100,7 +100,7 @@ class PipelineHookTriggerHandlerImpl extends AbstractWebHookTriggerHandler<Pipel
     protected CauseData retrieveCauseData(PipelineHook hook) {
         return causeData()
                 .withActionType(CauseData.ActionType.PIPELINE)
-                .withSourceProjectId(hook.getProjectId())
+                .withSourceProjectId(hook.getProject().getId())
                 .withBranch(getTargetBranch(hook)==null?"":getTargetBranch(hook))
                 .withSourceBranch(getTargetBranch(hook)==null?"":getTargetBranch(hook))
                 .withUserName(hook.getUser()==null||hook.getUser().getName()==null?"":hook.getUser().getName())
@@ -109,7 +109,7 @@ class PipelineHookTriggerHandlerImpl extends AbstractWebHookTriggerHandler<Pipel
                 .withSourceRepoSshUrl(hook.getRepository()==null||hook.getRepository().getGitSshUrl()==null?"":hook.getRepository().getGitSshUrl())
                 .withSourceRepoHttpUrl(hook.getRepository()==null||hook.getRepository()==null?"":hook.getRepository().getGitHttpUrl())
                 .withMergeRequestTitle("")
-                .withTargetProjectId(hook.getProjectId())
+                .withTargetProjectId(hook.getProject().getId())
                 .withTargetBranch(getTargetBranch(hook)==null?"":getTargetBranch(hook))
                 .withTargetRepoName("")
                 .withTargetNamespace("")
@@ -136,7 +136,7 @@ class PipelineHookTriggerHandlerImpl extends AbstractWebHookTriggerHandler<Pipel
     @Override
     protected BuildStatusUpdate retrieveBuildStatusUpdate(PipelineHook hook) {
         return buildStatusUpdate()
-            .withProjectId(hook.getProjectId())
+            .withProjectId(hook.getProject().getId())
             .withSha(hook.getObjectAttributes().getSha())
             .withRef(hook.getObjectAttributes().getRef())
             .build();

--- a/src/test/java/com/dabsquared/gitlabjenkins/trigger/handler/pipeline/PipelineHookTriggerHandlerImplTest.java
+++ b/src/test/java/com/dabsquared/gitlabjenkins/trigger/handler/pipeline/PipelineHookTriggerHandlerImplTest.java
@@ -70,7 +70,6 @@ public class PipelineHookTriggerHandlerImplTest {
 
         pipelineHookTriggerHandler = new PipelineHookTriggerHandlerImpl(allowedStates);
         pipelineHook = pipelineHook()
-            .withProjectId(1)
             .withUser(user)
             .withRepository(repository()
                 .withName("test")

--- a/src/test/resources/com/dabsquared/gitlabjenkins/webhook/build/PipelineEvent.json
+++ b/src/test/resources/com/dabsquared/gitlabjenkins/webhook/build/PipelineEvent.json
@@ -21,6 +21,7 @@
     "avatar_url": "http://www.gravatar.com/avatar/e32bd13e2add097461cb96824b7a829c?s=80\u0026d=identicon"
   },
   "project": {
+    "id": 1,
     "name": "Gitlab Test",
     "description": "Atque in sunt eos similique dolores voluptatem.",
     "web_url": "http://192.168.64.1:3005/gitlab-org/gitlab-test",

